bpf/lib/nodeport.h

```diff
@@ -2005,6 +2006,18 @@ static __always_inline int rev_nodeport_lb4(struct __ctx_buff *ctx, __u32 *ifind
 	tuple.saddr = ip4->saddr;
 
 	l4_off = l3_off + ipv4_hdrlen(ip4);
+
+#ifdef ENABLE_MASQUERADE
+	ret = ct_extract_ports4(ctx, l4_off, CT_EGRESS, &tuple);
+	if (ret == DROP_CT_UNKNOWN_PROTO)
+		goto skip_nat;
+	if (ret < 0)
+		return ret;
+
+	if (!ct_has_nodeport_egress_entry4(get_ct_map4(&tuple), &tuple))
+		goto skip_nat;
+#endif /* ENABLE_MASQUERADE */
```

Comment:
Thinking we might actually need to match the full condition in https://github.com/cilium/cilium/blob/4ddeb34a9192a3634d816a915c12629cac5340b7/bpf/bpf_host.c#L1196_L1201. And thus also add a similar check in the IPv6 path.

And bpf_overlay is even more [inclusive](https://github.com/cilium/cilium/blob/4ddeb34a9192a3634d816a915c12629cac5340b7/bpf/bpf_overlay.c#L563) in its SNAT usage :thinking: .

Reply:
> Thinking we might actually need to match the full condition in https://github.com/cilium/cilium/blob/4ddeb34a9192a3634d816a915c12629cac5340b7/bpf/bpf_host.c#L1196_L1201. And thus also add a similar check in the IPv6 path.
> 
Yeah, just checked. The CT entries for hostns-originating connections (which we track due to potential SNAT conflicts vs forwarded Nodeport connections) get the same RX double accounting. Will update.